### ISSUE TITLE
Filesystems not mounted

### ISSUE THREAD

**Issue by lnicola** [dasJ/sd-zfs#1]
I tried `sd-zfs`, but my root fs ended up in `/sysroot/pool/fs` instead of `/sysroot`, so switching the root didn't work. I'm using non-legacy mounts.

**Comment by lnicola** [comment]
Looks like my `mountpoint` was not set to `/` on my root filesystem. Not I can boot, but my other ZFS filesystems don't seem to get mounted, including `/home`, which makes things awkward. I'll try to look more into it.

**Comment by dasJ** [comment]
Does the current master work for you? I basically rewrote everything

**Comment by dasJ** [comment]
Sorry, the current AUR package will not work, I'll fix it asap

**Comment by lnicola** [comment]
All right, just ping me when it's fixed.

**Comment by dasJ** [comment]
Alright, can you test [this PKGBUILD](https://gist.github.com/dasJ/25bd695e4f4235d97c67dc5b20c204eb) (including the preset file)?

Please re-read the readme, it has changed a lot and is not completely compatible to the previous version.

**Comment by lnicola** [comment]
All right, so I booted with `root=zfs:pool/fs` and everything seems fine!

**Comment by dasJ** [comment]
So can this be closed?

**Comment by lnicola** [comment]
Only after you push the new PKGBUILD to AUR 😁.

**Comment by dasJ** [comment]
That's fair. I'll release the new version once my own machine can boot. I'm
currently having issues with encryption.

**Comment by dasJ** [comment]
@lnicola So I just pushed to the AUR, I hope everything works now :shipit:

**Comment by lnicola** [comment]
Just checked that it's fine, thanks for your work.

**Comment by obma** [comment]
Sorry for reopening, but it seems I'm running into the same/smiliar issue - with the newest version from the AUR installed (1.0.0-1; first download and install today).

I have a rootfs filesystem with mountpoint / and seperate filesystems for /home, /boot, /root, /srv and /var (non-legacy) - and a /boot.sd to boot the initramfs from.
On boot sysroot.mount fails. Importing the pool manually (zpool import -N mypool) and restarting sysroot.mount makes it possible to boot, but the other mountpoints aren't mounted then.

**Comment by dasJ** [comment]
@obma The problem is that mount(8) discards the output of the ZFS mounting tool.
Can you execute `mount.initrd_zfs` directly (in the recovery shell)?

For me systemd executes:

```
mount -t initrd_zfs tank/root /sysroot -o zfsutil
```

So I execute in the shell:

```
mount.initrd_zfs tank/root /sysroot -o zfsutil
```

Note I removed the -t parameter.

**Comment by obma** [comment]
I switched back to not using systemd in the initrd and use the zfs hook for now (as I do on my older machines)  - `HOOKS="base udev autodetect modconf block keyboard zfs filesystems"`

But I can test your suggestion on the weekend - if it helps you figuring out what's going on there.

**Comment by dasJ** [comment]
Thank you, I really hope I can fix it 😃

**Comment by dasJ** [comment]
Closing this for now. Please open a new issue once you could further investigate your problem. The cause of your problem is another than the cause of the OP.
